### PR TITLE
Encrypt grpc connection

### PR DESCRIPTION
**Pull request by n-hutton**

### CODE REVIEW

File: docker/ml.Dockerfile [hunk 1]
```diff
@@ -21,15 +21,10 @@ RUN cd ./colearn && \
     pip3 install --no-cache-dir --upgrade pip && \
     pip3 install --no-cache-dir -e .[all]
 
-RUN groupadd -g 999 appuser && \
-    useradd -r -u 999 -g appuser appuser
-
-USER appuser
```
Comment (n-hutton) [inline docker/ml.Dockerfile:27]: Since we need to add certificate etc. I removed the adduser functionality

File: colearn_grpc/grpc_server.py [hunk 2]
```diff
@@ -56,12 +57,37 @@ def run(self):
...
 
+        # There needs to be a certificate and private key available to enable encryption -
+        # if these are not available, fall back to no encryption.
+        encrypted_connection = True
+
+        if not os.path.isfile("server.crt"):
+            _logger.error("Failed to find file server.crt needed for encrypted grpc connection - not enabling")
+            encrypted_connection = False
+
+        if not os.path.isfile("server.key"):
```
Comment (evsmithx) [inline colearn_grpc/grpc_server.py:68]: These filenames are used twice and could be in constants. Also a path that is relative to the repo root would be better!

Comment (evsmithx) [reply colearn_grpc/grpc_server.py:68]: E.g.
```python
REPO_ROOT = Path(__file__).absolute().parent.parent
server_key = REPO_ROOT / "server.key"
```

**Review by evsmithx — APPROVED** [review]
More minor comments, and the same question as in contract-learn: if the client tries to use `insecure_connection` when the connection is encrypted, will this fail? Because that will be a problem in `example_grpc_client.py`

File: colearn_grpc/grpc_server.py (hunk 2 above)
Comment (n-hutton) [reply colearn_grpc/grpc_server.py]: Not twice? server.crt and server.key. But you are right, it would be cleaner to have them as constants etc.

**Review by evsmithx — APPROVED** [review]

**Pull request merged**